modules/plugins/utility/motion/precognition/precognition.nix

```diff
@@ -0,0 +1,78 @@
+{lib, ...}: let
+  inherit (lib.options) mkEnableOption mkOption literalExpression;
+  inherit (lib.types) attrsOf listOf str bool int submodule;
+in {
+  options.vim.utility.motion.precognition = rec {
+    enable = mkEnableOption "precognition.nvim plugin";
+
+    startVisible = mkOption {
+      type = bool;
+      description = "Whether to start 'precognition' automatically.";
+      default = true;
+    };
+
+    showBlankVirtLine = mkOption {
+      type = bool;
+      description = "Whether to show a blank virtual line when no movements are shown.";
+      default = true;
+    };
+
+    highlightColor = mkOption {
+      type = attrsOf str;
+
+      example = literalExpression ''
+        { link = "Comment"; }
+        # or
+        { foreground = "#0000FF"; background = "#000000"; };
+      '';
+      default = {link = "Comment";};
```

Comment:
```suggestion
    highlightColor = mkOption {
      type = attrsOf str;
      default = {link = "Comment";};
      example = literalExpression ''
        { link = "Comment"; }
        # or
        { foreground = "#0000FF"; background = "#000000"; };
      '';
```

Reply:
Resolved in e43b18cb9f3bd41fa1edf9823493073ddc859d7b.